### ISSUE TITLE
Docker build

### ISSUE THREAD

**Issue by oxyc** [geerlingguy/packer-centos-7#10]
_This is actually in reference to Drupal VM but I thought I'd post about it in one of these box repositories instead. Seems like the more appropriate place. Also I havent read up on packer/docker/vagrant integrations at all, I just saw that it's possible to use docker with Vagrant._

Would it be a good idea to provide "supported" docker builds for these boxes as well? I'm imagining using those as bases for all your role tests on travis, as well as possibly even running Drupal VM within a container instead of a VM. Sounds pretty cool for local CI setups.

I'm not saying this would be provided out of the box in Drupal VMs case, but if we facilitate working boxes we can add a section to the docs about how to use containers through `Vagrantfile.local`.

I have no idea if this is even feasible or maintainable though... It just sounds awesome.

**Comment by geerlingguy** [comment]
The only issue for me is that the overwhelming majority of Drupal VM users (and heck, most other dev environment projects I've worked with) are running in a Mac OS X or Windows environment—neither of which natively support Docker.

So even if I did go the Docker route, it would have to be either wrapped in something like Boot2Docker or a VM in general, meaning I'd just be digging the local environment abstraction one level deeper.

I'm not opposed to moving some portions of Drupal VM inside containers inside the VM if it adds flexibility, speed, or agility—for example, moving a service like Varnish, or Memcached, or Solr into a container could make for cleaner configuration separation.

However, I'm leery of doing things like wrapping PHP+nginx, PHP+apache, or PHP-FPM and some other cruft in one giant Docker-container-like-VM, and I have never been that happy with MySQL inside a Docker container, especially if it's not for any performance or portability reason.

I have other reservations about going down the Docker route too much further for a project as flexible/sprawling as Drupal VM, and I also know there are already dozens if not hundreds of excellent base images in Docker Hub for PHP, Drupal, Apache, Nginx, MySQL, etc.

Most of the developers who dive headfirst into Docker are willing to put the effort into composing their own project-specific environments, and I don't know if trying to replicate the Drupal VM 'system in a box' approach using Docker would appeal to those developers.

Basically what I'm saying is, if Docker makes Drupal VM maintenance and architecture easier, then I'm going to use it. So far that hasn't been the case in my dabbling, but I'm always willing to change :)

**Comment by bhundven** [comment]
I know this was a while ago, but:

https://docs.docker.com/engine/installation/mac/

https://docs.docker.com/engine/installation/windows/

**Comment by geerlingguy** [comment]
@bhundven - Once the official Mac/Windows clients are out of beta, I'm thinking of going back and re-evaluating the situation. The current tooling (using docker-machine) is incredibly painful if you're not on a Linux machine that natively supports Docker.

**Comment by geerlingguy** [comment]
Docker is still in beta, but now the beta is open for anyone on Mac/Windows. It would be interesting to take a look at this sometime.

**Comment by geerlingguy** [comment]
Just wanted to post that in the intervening months, Docker has been released, and I do now maintain a number of Ansible-based-base images on Docker Hub: https://hub.docker.com/u/geerlingguy/

**Comment by bhundven** [comment]
+1

I've been using the [docker toolbox](https://www.docker.com/products/docker-toolbox) on Mac OS X/macOS with minikube for a month, and it seems to work pretty well!
